cli/command/service/progress/progress.go

```diff
@@ -0,0 +1,378 @@
+package progress
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/pkg/stringid"
+	"golang.org/x/net/context"
+)
+
+var (
+	stateToProgress = map[swarm.TaskState]int64{
+		swarm.TaskStateNew:       1,
+		swarm.TaskStateAllocated: 2,
+		swarm.TaskStatePending:   3,
+		swarm.TaskStateAssigned:  4,
+		swarm.TaskStateAccepted:  5,
+		swarm.TaskStatePreparing: 6,
+		swarm.TaskStateReady:     7,
+		swarm.TaskStateStarting:  8,
+		swarm.TaskStateRunning:   9,
+	}
+
+	longestState = func() int {
+		longest := 0
+		for state := range stateToProgress {
+			if len(state) > longest {
+				longest = len(state)
+			}
+		}
+		return longest
```

Comment:
Minor: I usually see this kind of code in a `func init()` instead of `var`. I'm not sure if there is any real difference or if it's just best practice.